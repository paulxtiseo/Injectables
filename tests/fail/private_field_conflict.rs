use injectables::{injectable, inject_fields};

#[injectable]
pub struct Base {
  id: u64,  // private field
}

#[injectable]
pub struct Derived {
  pub id: u64,  // public field with same name
}

#[inject_fields(Base, Derived)]
pub struct Conflict {
  pub name: String,
}

fn main() {}