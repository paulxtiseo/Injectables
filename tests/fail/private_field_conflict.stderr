error: Conflicting visibility for field 'id': cannot have both private and public fields with the same name
  --> tests/fail/private_field_conflict.rs:13:1
   |
13 | #[inject_fields(Base, Derived)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `inject_fields` (in Nightly builds, run with -Z macro-backtrace for more info)
